src/civibuild.lib.sh

```diff
@@ -472,7 +472,9 @@ function civicrm_install() {
 
   pushd "$CIVI_CORE" >> /dev/null
     ## Does this build include development support (eg git or tarball-based)?
-    if [ -e "xml" -a -e "bin/setup.sh" ]; then
+    if [ "$NO_SAMPLE_DATA" == "1" ]; then
```

Comment:
OK, makes sense that you'd call `setup.sh` with different args for this scenario.

With this patch, the `if`/`elif` chain is now addressing two distinct questions. These feel orthogonal from a user perspective, but each combination needs a slightly different implementation. The two questions are:

 * Are we working from raw source (i.e. using `git` to download `xml/schema/` and run `bin/setup.sh`) or from a published tarball (i.e. there's no `xml/schema/` but instead we have files in `sql/`)?
 * Do we want an empty/pristine DB or a populated/demo DB?

which leads to four use-cases:

 1. Have raw source (`-e xml -a -e bin/setup.sh`) and want sample data (`-z "$NO_SAMPLE_DATA"`)
 2. Have raw source (`-e xml -a -e bin/setup.sh`) and do not want sample data (`-n "$NO_SAMPLE_DATA"`)
 3. Have tarballs (`-e sql/civicrm.mysql`) files and want sample data (`-z "$NO_SAMPLE_DATA"`)
 4. Have tarballs (`-e sql/civicrm.mysql`) and do not want sample data (`-n "$NO_SAMPLE_DATA"`)

The current patch is going to confuse cases 2 and 4. One fix might be adding more conditionals and another branch in the `if`/`elif` chain.

Note: For case 4, the install logic should be `cat sql/civicrm.mysql sql/civicrm_data.mysql | eval mysql $CIVI_DB_ARGS`